tests/functional/aws-node-sdk/test/versioning/objectCopy.js

```diff
@@ -0,0 +1,1006 @@
+import assert from 'assert';
+import async from 'async';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+import { removeAllVersions } from '../../lib/utility/versioning-util';
+
+const sourceBucketName = 'supersourcebucket8102016';
+const sourceObjName = 'supersourceobject';
+const destBucketName = 'destinationbucket8102016';
+const destObjName = 'copycatobject';
+
+const originalMetadata = {
+    oldmetadata: 'same old',
+    overwriteme: 'wipe me out with replace',
+};
+const originalCacheControl = 'max-age=1337';
+const originalContentDisposition = 'attachment; filename="1337.txt";';
+const originalContentEncoding = 'base64,aws-chunked';
+const originalExpires = new Date(12345678);
+
+const newMetadata = {
+    newmetadata: 'new kid in town',
+    overwriteme: 'wiped',
+};
+const newCacheControl = 'max-age=86400';
+const newContentDisposition = 'attachment; filename="fname.ext";';
+const newContentEncoding = 'gzip,aws-chunked';
+const newExpires = new Date();
+
+const content = 'I am the best content ever';
+
+const otherAccountBucketUtility = new BucketUtility('lisa', {});
+const otherAccountS3 = otherAccountBucketUtility.s3;
+
+function checkNoError(err) {
+    assert.equal(err, null,
+        `Expected success, got error ${JSON.stringify(err)}`);
+}
+
+function checkError(err, code) {
+    assert.notEqual(err, null, 'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+}
+
+function dateFromNow(diff) {
+    const d = new Date();
+    d.setHours(d.getHours() + diff);
+    return d.toISOString();
+}
+
+function dateConvert(d) {
+    return (new Date(d)).toISOString();
+}
+
+const testing = process.env.VERSIONING === 'no' ? describe.skip : describe;
+
+testing('Object Version Copy', () => {
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+        let etag;
+        let etagTrim;
+        let lastModified;
+        let versionId = undefined;
+        let copySource = undefined;
+
+        before(() => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
```

Comment:
It's odd to see cleanup of resources in `before()`, it should be handled in `after()`